protocol/avdl/keybase1/implicit_team_upgrade.avdl

```diff
@@ -0,0 +1,7 @@
+@namespace("keybase.1")
+
+// implicitTeamUpgrade defines a protocol that service calls into KBFS for
+// marking the start of migrating a TLF into iTeam backed form.
+protocol implicitTeamUpgrade {
+    void upgradeLock(string tlfName);
```

Comment:
Also, it's pretty confusing that this protocol/RPC uses the word "upgrade", but the mdserver one uses the word "migrate".  Can we be consistent?  "migrate"/"migration" seems a bit clearer to me.  E.g. this can be:

```go
protocol implicitTeamMigration {
    void startMigration(Folder folder);
}
```